src/components/CurrencyInputPanel/CurrencyInputPanel.component.tsx

```diff
@@ -74,6 +76,28 @@ export const CurrencyInputPanelComponent = ({
     }
   }, [disableCurrencySelect, isLoading])
 
+  const handleOnMax = useCallback(() => {
+    if (onMax && setIsMaxAmount) {
+      onMax()
+      setIsMaxAmount(true)
+    }
+  }, [onMax, setIsMaxAmount])
```

Comment:
You can remove `setIsMaxAmount` from dependencies. `setState` will not change on re-renders

Reply:
Addressed in [60ad9c7](https://github.com/levelkdev/swapr-dapp/pull/1013/commits/60ad9c7e9feff96ac053e0555e59897acb3af6fd). Thanks!